clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -896,11 +896,11 @@ public void assign(Collection<TopicPartition> partitions) {
 
         log.info("Assigned to partition(s): {}", join(partitions, ", "));
         if (this.subscriptions.assignFromUser(new HashSet<>(partitions)))
-           updateMetadata(time.milliseconds());
+           updateMetadata();
     }
 
-    private void updateMetadata(long milliseconds) {
-        final MetadataUpdateApplicationEvent event = new MetadataUpdateApplicationEvent(milliseconds);
+    private void updateMetadata() {
+        final NewTopicsMetadataUpdateRequestEvent event = new NewTopicsMetadataUpdateRequestEvent();
```

Comment:
Just for my understanding, why can't we rely here on the existing metadata object with the `metadata.requestUpdateForNewTopics` call? I see that's still used for this same purpose in other places like the PrototypeAsyncConsumer.subscribe, but maybe I missing the reason for this and then is the subscribe that might need to be updated.

Reply:
hmm good point.  Originally I wanted metadata only owned by the background thread so that the ownership is clearer, which also allows us to remove the synchronization blocks.

Reply:
we wanted to remove as much concurrent access to objects as possible.  It wasn't necessarily for performance reasons, I think it was more for removing the complexity of the code.  I think the synchronization does have a performance impact, though that's not really the point of the refactoring.